src/connection.js

```diff
@@ -986,6 +1023,7 @@ class Connection extends EventEmitter {
       }
 
       this.request = request;
+      this.request.connection = this;
```

Comment:
see my comment in #536.

And it wouldn't be enough to set only a reference to the function, because you also need the context, e.g.:

`this.request.pauseCallback = () => {this.pauseRequest(); };`

And then the question arises, whether Request.pauseCallback is a public member of the Request class. You have more or less the same ugliness and more confusion.